components/style/properties/longhand/ui.mako.rs

```diff
@@ -23,3 +23,59 @@ ${helpers.single_keyword("-moz-user-select", "auto text none all", products="gec
                          gecko_inexhaustive=True,
                          animatable=False,
                          spec="https://drafts.csswg.org/css-ui-4/#propdef-user-select")}
+
+<%helpers:longhand name="caret-color" animatable="False" products="none"
+                   spec="https://drafts.csswg.org/css-ui/#caret-color">
+    use cssparser::Color;
+    use std::fmt;
+    use style_traits::ToCss;
+    use values::NoViewportPercentage;
+    use values::computed::ComputedValueAsSpecified;
+    pub use self::computed_value::T as SpecifiedValue;
+
+    impl NoViewportPercentage for SpecifiedValue {}
```

Comment:
After #15411, we need to use `no_viewport_percentage!(SpecifiedValue);` here instead.